fastlane/lib/fastlane/actions/jira.rb

```diff
@@ -0,0 +1,83 @@
+        password     = params[:password]
+        ticket_id    = params[:ticket_id]
+        comment_text = params[:comment_text]
+
+        options = {
+                    site: site,
+                    context_path: context_path,
+                    auth_type: auth_type,
+                    username: username,
+                    password: password
+                  }
+
+        client = JIRA::Client.new(options)
+        issue = client.Issue.find(ticket_id)
+        comment = issue.comments.build
+        comment.save({ 'body' => comment_text })
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Leave a comment on JIRA tickets"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :url,
+                                      env_name: "FL_JIRA_SITE",
+                                      description: "URL for Jira instance",
+                                       verify_block: proc do |value|
+                                         raise "No url for Jira given, pass using `url: 'url'`".red unless value and !value.empty?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :username,
+                                       env_name: "FL_JIRA_USERNAME",
+                                       description: "Username for JIRA instance",
+                                       verify_block: proc do |value|
+                                         raise "No username".red unless value and !value.empty?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :password,
+                                       env_name: "FL_JIRA_PASSWORD",
+                                       description: "Password for Jira",
+                                       verify_block: proc do |value|
+                                         raise "No password".red unless value and !value.empty?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :ticket_id,
+                                       env_name: "FL_JIRA_TICKET_ID",
+                                       description: "Ticket ID for Jira, i.e. IOS-123",
+                                       verify_block: proc do |value|
+                                         raise "No Ticket specified".red unless value and !value.empty?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :comment_text,
+                                       env_name: "FL_JIRA_COMMENT_TEXT",
+                                       description: "Text to add to the ticket as a comment",
+                                       verify_block: proc do |value|
+                                         raise "No comment specified".red unless value and !value.empty?
```

Comment:
Instead of using `raise` use the following:
```ruby
UI.user_error!("No comment specified") if value.to_s.length == 0
```
also note how we check if a string is empty